### ISSUE TITLE
Stopping on Docker provider should remove the containers

### ISSUE THREAD

**Issue by rtnpro** [projectatomic/atomicapp#389]
# Issue

Currently, stopping a Nulecule application on Docker provider just stops the running containers. If you have named your containers, an attempt to run the Nulecule application, for the second time, will fail because the names of the containers to be run will collide with the stopped containers from the previous run.

# Solution
Instead of just stopping the Docker containers, we should also remove them.

**Comment by dustymabe** [comment]
So one thing about this is that we are planning to make other steps in the future (see https://github.com/dustymabe/atomicapp/commit/bacf8ac3eb0fdee7462199887bec3b5f74411312). I think the action of removing the containers would be part of another step (not stop). 

Another thing is that we should really re-evaluate having "named" containers. If we need named containers for docker links then we could have nulecule itself come up with random names for the containers to then plug in to the links. 

I think Docker 1.9 has basic service discovery where links is possibly no longer needed. Would need some investigation.

**Comment by rtnpro** [comment]
Yes, cleaning up containers should go to a different step, and that's in our roadmap. However, this issue leaves the current user experience using Docker provider poor.

The main culprit for this issue is sharing to use hard coded names for docker linking. We have discussed in the past about some way to generate run time data when deploying artifacts, in this case, container ID, name, etc. This won't make much a value addition in case of ``k8s`` and openshift, but it can be a useful feature if someone wants to use it.

**Noted**: I will need to look into service discovery in Docker 1.9.

@dustymabe shall I work on a POC for atomicapp to keep track of values generated during runtime for use in subsequent steps?

**Comment by dustymabe** [comment]
@rtnpro I'd say for now let's concentrate on beta3 testing/bugfix and revisit this subject next week. Sound reasonable?

**Comment by rtnpro** [comment]
@dustymabe yeah!

**Comment by dustymabe** [comment]
talked with rtnpro today. we agreed that this should/could be handled by another command rather than stop.

**Comment by rtnpro** [comment]
@dustymabe I am implementing ``clean`` command for it, ok?

**Comment by dustymabe** [comment]
We had to re-open because of issues with stop on docker provider. see comments in #457 for details.

**Comment by dustymabe** [comment]
Removing the blocker label from this. We won't be doing this before GA because:
 
- There are issues with removing a docker container from within atomicapp prior to docker 1.9
